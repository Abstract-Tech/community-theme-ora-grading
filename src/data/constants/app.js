import { getConfig } from '@edx/frontend-platform';

// eslint-disable-next-line import/prefer-default-export
export const routePath = `${getConfig().PUBLIC_PATH}:courseId`;
export const locationId = window.location.pathname.slice(1)