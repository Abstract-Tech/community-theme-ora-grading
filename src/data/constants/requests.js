import { StrictDict } from 'utils';

export const RequestStates = StrictDict({
  inactive: 'inactive',
  pending: 'pending',
  completed: 'completed',
  failed: 'failed',
});

export const RequestKeys = StrictDict({
  initialize: 'initialize',
  fetchSubmission: 'fetchSubmission',
  fetchSubmissionStatus: 'fetchSubmissionStatus',
  setLock: 'setLock',
  prefetchNext: 'prefetchNext',
  prefetchPrev: 'prefetchPrev',
  submitGrade: 'submitGrade',
});
