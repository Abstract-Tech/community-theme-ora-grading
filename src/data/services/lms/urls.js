import { StrictDict } from 'utils';
import { configuration } from 'config';

const baseUrl = `${configuration.LMS_BASE_URL}`;

const api = `${baseUrl}/api/`;
const course = (courseId) => `${baseUrl}/courses/${courseId}`;

const openResponse = (courseId) => (
  `${course(courseId)}/instructor#view-open_response_assessment`
);
const ora = (courseId, locationId) => `${course(courseId)}/jump_to/${locationId}`;

export default StrictDict({
  api,
  baseUrl,
  course,
  openResponse,
  ora,
});
