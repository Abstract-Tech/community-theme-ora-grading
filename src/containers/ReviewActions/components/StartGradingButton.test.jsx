import React from 'react';
import { shallow } from 'enzyme';

import selectors from 'data/selectors';
import thunkActions from 'data/thunkActions';
import { gradingStatuses as statuses } from 'data/services/lms/constants';

import {
  StartGradingButton,
  mapStateToProps,
  mapDispatchToProps,
} from './StartGradingButton';

jest.mock('@edx/paragon', () => ({
  Button: () => 'Button',
}));
jest.mock('@edx/paragon/icons', () => ({
  Cancel: 'Cancel',
  Highlight: 'Highlight',
}));
jest.mock('data/selectors', () => ({
  __esModule: true,
  default: {
    grading: {
      selected: {
        gradingStatus: (state) => ({ gradingStatus: state }),
      },
    },
  },
}));

describe('StartGradingButton component', () => {
  describe('component', () => {
    const props = {};
    beforeEach(() => {
      props.startGrading = jest.fn().mockName('this.props.startGrading');
      props.stopGrading = jest.fn().mockName('this.props.stopGrading');
    });
    const render = (gradingStatus) => shallow(
      <StartGradingButton {...props} gradingStatus={gradingStatus} />,
    );
    test('snapshot: locked (null)', () => {
      const el = render(statuses.locked);
      expect(el).toMatchSnapshot();
      expect(el).toEqual({});
    });
    test('snapshot: ungraded (startGrading callback)', () => {
      expect(render(statuses.ungraded)).toMatchSnapshot();
    });
    test('snapshot: graded (startGrading callback', () => {
      expect(render(statuses.graded)).toMatchSnapshot();
    });
    test('snapshot: inProgress (stopGrading callback', () => {
      expect(render(statuses.inProgress)).toMatchSnapshot();
    });
  });
  describe('mapStateToProps', () => {
    let mapped;
    const testState = { some: 'test-state' };
    beforeEach(() => {
      mapped = mapStateToProps(testState);
    });
    test('gradingStatus loads from grading.selected.gradingStatus', () => {
      expect(mapped.gradingStatus).toEqual(selectors.grading.selected.gradingStatus(testState));
    });
  });
  describe('mapDispatchToProps', () => {
    it('loads startGrading from thunkActions.grading.stargGrading', () => {
      expect(mapDispatchToProps.startGrading).toEqual(thunkActions.grading.startGrading);
    });
    it('loads stopGrading from thunkActions.grading.stopGrading', () => {
      expect(mapDispatchToProps.stopGrading).toEqual(thunkActions.grading.stopGrading);
    });
  });
});
