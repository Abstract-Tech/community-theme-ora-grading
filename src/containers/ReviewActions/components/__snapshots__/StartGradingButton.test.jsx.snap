// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StartGradingButton component component snapshotes snapshot: graded, confirmOverride (startGrading callback) 1`] = `
<React.Fragment>
  <Button
    iconAfter="Highlight"
    onClick={[MockFunction this.handleClick]}
    variant="primary"
  >
    Override grade
  </Button>
  <OverrideGradeConfirmModal
    isOpen={true}
    onCancel={[MockFunction this.hideConfirmOverrideGrade]}
    onConfirm={[MockFunction this.confirmOverrideGrade]}
  />
  <StopGradingConfirmModal
    isOpen={false}
    isOverride={true}
    onCancel={[MockFunction this.hideConfirmStopGrading]}
    onConfirm={[MockFunction this.confirmStopGrading]}
  />
</React.Fragment>
`;

exports[`StartGradingButton component component snapshotes snapshot: inProgress, isOverride, confirmStop (stopGrading callback) 1`] = `
<React.Fragment>
  <Button
    iconAfter="Cancel"
    onClick={[MockFunction this.handleClick]}
    variant="primary"
  >
    Stop grading this response
  </Button>
  <OverrideGradeConfirmModal
    isOpen={false}
    onCancel={[MockFunction this.hideConfirmOverrideGrade]}
    onConfirm={[MockFunction this.confirmOverrideGrade]}
  />
  <StopGradingConfirmModal
    isOpen={true}
    isOverride={true}
    onCancel={[MockFunction this.hideConfirmStopGrading]}
    onConfirm={[MockFunction this.confirmStopGrading]}
  />
</React.Fragment>
`;

exports[`StartGradingButton component component snapshotes snapshot: locked (null) 1`] = `null`;

exports[`StartGradingButton component component snapshotes snapshot: ungraded (startGrading callback) 1`] = `
<React.Fragment>
  <Button
    iconAfter="Highlight"
    onClick={[MockFunction this.handleClick]}
    variant="primary"
  >
    Start Grading
  </Button>
  <OverrideGradeConfirmModal
    isOpen={false}
    onCancel={[MockFunction this.hideConfirmOverrideGrade]}
    onConfirm={[MockFunction this.confirmOverrideGrade]}
  />
  <StopGradingConfirmModal
    isOpen={false}
    isOverride={false}
    onCancel={[MockFunction this.hideConfirmStopGrading]}
    onConfirm={[MockFunction this.confirmStopGrading]}
  />
</React.Fragment>
`;
